ts/Dashboards/Components/Component.ts

```diff
@@ -838,6 +870,32 @@ abstract class Component {
         return this;
     }
 
+    /**
+     * Removes the component instance from the provided connector.
+     * Stops the connector polling if the last element is removed.
+     *
+     * @param connector
+     * Instance of a connector.
+     *
+     * @internal
+     */
+    public removeConnectorAssignment(
+        connector: Component.ConnectorTypes | undefined
+    ): void {
+        const component = this;
+        if (connector && connector.elements.includes(component.element)) {
```

Comment:
1. You can use optional `connector?.elements.includes`
2. `connector.elements` appears a few times, consider using `const ` and assign a reference.

Reply:
`connector &&` makes sure it's defined inside. When I do it this way, it shows errors on every usage. My point is I should either use the optional chaining or define the connector somewhere anyway. If I then don't follow this suggestion, there's only one usage of `connector.elements` I could use a variable for.